core/services/ocr2/plugins/ocr2keeper/evm21/mercury/streams/streams.go

```diff
@@ -0,0 +1,327 @@
+			checkResults[i].IneligibilityReason = uint8(reason)
+			checkResults[i].Retryable = retryable
+			return
+		} else if !allowed {
+			lookupLggr.Debugf("at block %d upkeep %s NOT allowed to query Mercury server", block, upkeepId)
+			checkResults[i].IneligibilityReason = uint8(mercury.MercuryUpkeepFailureReasonMercuryAccessNotAllowed)
+			return
+		}
+	} else if streamsLookupResponse.IsMercuryVersionUnkown() {
+		// if mercury version cannot be determined, set failure reason
+		lookupLggr.Debugf("at block %d upkeep %s NOT allowed to query Mercury server", block, upkeepId)
+		checkResults[i].IneligibilityReason = uint8(mercury.MercuryUpkeepFailureReasonInvalidRevertDataInput)
+		return
+	}
+
+	streamsLookupResponse.UpkeepId = upkeepId.BigInt()
+	// the block here is exclusively used to call checkCallback at this block, not to be confused with the block number
+	// in the revert for mercury v0.2, which is denoted by time in the struct bc starting from v0.3, only timestamp will be supported
+	streamsLookupResponse.Block = uint64(block.Int64())
+	lookupLggr.Infof("at block %d upkeep %s DecodeStreamsLookupRequest feedKey=%s timeKey=%s feeds=%v time=%s extraData=%s", block, upkeepId, streamsLookupResponse.FeedParamKey, streamsLookupResponse.TimeParamKey, streamsLookupResponse.Feeds, streamsLookupResponse.Time, hexutil.Encode(streamsLookupResponse.ExtraData))
+	lookups[i] = streamsLookupResponse
+}
+
+func (s *streams) doLookup(ctx context.Context, wg *sync.WaitGroup, lookup *mercury.StreamsLookup, i int, checkResults []ocr2keepers.CheckResult) {
+	defer wg.Done()
+
+	state, reason, values, retryable, retryInterval, err := mercury.NoPipelineError, mercury.MercuryUpkeepFailureReasonInvalidRevertDataInput, [][]byte{}, false, 0*time.Second, fmt.Errorf("invalid revert data input: feed param key %s, time param key %s, feeds %s", lookup.FeedParamKey, lookup.TimeParamKey, lookup.Feeds)
+	pluginRetryKey := generatePluginRetryKey(checkResults[i].WorkID, lookup.Block)
+
+	if lookup.IsMercuryV02() {
+		state, reason, values, retryable, retryInterval, err = s.v02Client.DoRequest(ctx, lookup, pluginRetryKey)
+	} else if lookup.IsMercuryV03() {
+		state, reason, values, retryable, retryInterval, err = s.v03Client.DoRequest(ctx, lookup, pluginRetryKey)
+	}
+
+	if err != nil {
+		s.lggr.Errorf("upkeep %s retryable %v retryInterval %s doMercuryRequest: %s", lookup.UpkeepId, retryable, retryInterval, err.Error())
+		checkResults[i].Retryable = retryable
+		checkResults[i].RetryInterval = retryInterval
+		checkResults[i].PipelineExecutionState = uint8(state)
+		checkResults[i].IneligibilityReason = uint8(reason)
+		return
+	}
+
+	for j, v := range values {
+		s.lggr.Infof("upkeep %s doMercuryRequest values[%d]: %s", lookup.UpkeepId, j, hexutil.Encode(v))
+	}
+
+	state, retryable, mercuryBytes, err := s.checkCallback(ctx, values, lookup)
+	if err != nil {
+		s.lggr.Errorf("at block %d upkeep %s checkCallback err: %s", lookup.Block, lookup.UpkeepId, err.Error())
```

Comment:
instead of `at block` can you change this to `at time param key`?
both v0.2 and v0.3 code path may reach here.